RecoParticleFlow/PFTracking/src/PFTrackAlgoTools.cc

```diff
@@ -237,9 +237,11 @@ namespace PFTrackAlgoTools {
                         const std::vector<double>& DPtovPtCut,
                         const std::vector<unsigned>& NHitCut,
                         bool useIterTracking,
-                        bool debug) {
-    //recheck that the track is high purity!
-    if (!trackref->quality(reco::TrackBase::highPurity))
+                        bool debug,
+			std::string trackQuality) {
+    
+    //check quality of tracks
+    if(!trackref->quality(reco::TrackBase::qualityByName(trackQuality)))
```

Comment:
Actually, it would be better to do the string-to-enum conversion (`reco::TrackBase::qualityByName()`) once in the constructor (and possibly throw an exception if the returned value is `undefQuality`), and pass the enum to this function.

Reply:
Changing the `trackQuality` parameter here and `GeneralTracksImporter::trackQuality_` to `TrackBase::TrackQuality` enum? Yes, please do (and of course it needs to be eventually tested).